pgxmock.AddRows makes []bytes conversion panic

**Describe the bug**
I applied pgx.CollectRows and pgx.RowToAddrOfStructByPos to simplify sql query logic.
Like this:
`pgx.CollectRows(rows, pgx.RowToAddrOfStructByPos[{{mymodel struct}}]`
Then I tested the function logic which contains above line, it panicked and a message was:
`panic: interface conversion: interface {} is string, not []uint8`

**To Reproduce**
Steps to reproduce the behavior:
1. Just I described bug, you can reproduce when you use pgx.CollectRows and pgx.RowToaddrOfStructByPos
2. Then use pgxmock.AddRow in the test code.

**Expected behavior**
Not panicking

**Screenshots**
<img width="620" alt="image" src="https://user-images.githubusercontent.com/42539506/223687579-8dc94d5f-3ab1-4645-8d49-1efd69fedac3.png">


**Desktop (please complete the following information):**
 - OS: macOS Monterey 12.5.1
 - Browser chrome
 - Versio 108.0.5359.124 (Official Build) (arm64)

**Smartphone (please complete the following information):**
 - Device: iPhone 14 Pro
 - OS: iOS 16.1
 - Browser chrome
 - Version Version 110.0.5481.114

**Additional context**
Conversion to bytes slice with considering type would resolve this issue. I will fix this issue when I have some time to get it. Thank you.

I understand your hardship. I cannot provide raw code snippet, but I'm going to show you similar a code snippet.
`func ListModel(ctx context.Context, id string) ([]*Model, error) {
	query := "SELECT * FROM model WHERE id = $1"
	rows, err := sql.DB.Query(ctx, query, id)
	if err != nil {
		return nil, err
	}
	defer rows.Close()

	models, err := pgx.CollectRows(rows, pgx.RowToAddrOfStructByPos[Model])
	if err != nil {
		return nil, sql.HandleError(err)
	}

	~~~
}`

And I test it in my unit test code with using pgxmock's expect query.

`mockDB.ExpectQuery("SELECT [*] FROM model WHERE id = [$]1").
			WithArgs("id").
			WillReturnRows(pgxmock.NewRows([]string{"~~ some model columns~~~"}).
				AddRow(some rows~~~))
`

And its error log was just like this:
panic: interface conversion: interface {} is string, not []uint8 [recovered]
	panic: interface conversion: interface {} is string, not []uint8

I understand your hardship. I cannot provide raw code snippet, but I'm going to show you similar a code snippet.
`func ListModel(ctx context.Context, id string) ([]*Model, error) {
	query := "SELECT * FROM model WHERE id = $1"
	rows, err := sql.DB.Query(ctx, query, id)
	if err != nil {
		return nil, err
	}
	defer rows.Close()

	models, err := pgx.CollectRows(rows, pgx.RowToAddrOfStructByPos[Model])
	if err != nil {
		return nil, err
	}

	~~~
}`

And I test it in my unit test code with using pgxmock's expect query.

`mockDB.ExpectQuery("SELECT [*] FROM model WHERE id = [$]1").
			WithArgs("id").
			WillReturnRows(pgxmock.NewRows([]string{"~~ some model columns~~~"}).
				AddRow(some rows~~~))
`

And its error log was just like this:
panic: interface conversion: interface {} is string, not []uint8 [recovered]
	panic: interface conversion: interface {} is string, not []uint8

--- a/rows.go
+++ b/rows.go
@@ -127,7 +127,12 @@ func (rs *rowSets) RawValues() [][]byte {
 			dest[i] = b
 			continue
 		}
-		dest[i] = col.([]byte)
+		d, ok := col.([]byte)
+		if ok {
+			dest[i] = d
+		} else {
+			dest[i] = []byte(fmt.Sprintf("%v", col))
+		}
 	}
 
 	return dest

--- a/rows_test.go
+++ b/rows_test.go
@@ -848,3 +848,56 @@ func TestEmptyRowSets(t *testing.T) {
 // 		t.Fatal(err)
 // 	}
 // }
+
+func TestMockQueryWithCollect(t *testing.T) {
+	t.Parallel()
+	mock, err := NewConn()
+	if err != nil {
+		t.Errorf("an error '%s' was not expected when opening a stub database connection", err)
+	}
+	defer mock.Close(context.Background())
+	type rowStructType struct {
+		ID    int
+		Title string
+	}
+	rs := NewRows([]string{"id", "title"}).AddRow(5, "hello world")
+
+	mock.ExpectQuery("SELECT (.+) FROM articles WHERE id = ?").
+		WithArgs(5).
+		WillReturnRows(rs)
+
+	rows, err := mock.Query(context.Background(), "SELECT (.+) FROM articles WHERE id = ?", 5)
+	if err != nil {
+		t.Errorf("error '%s' was not expected while retrieving mock rows", err)
+	}
+
+	defer rows.Close()
+
+	//if !rows.Next() {
+	//	t.Error("it must have had one row as result, but got empty result set instead")
+	//}
+
+	rawMap, err := pgx.CollectRows(rows, pgx.RowToStructByPos[rowStructType])
+	if err != nil {
+		t.Errorf("error '%s' was not expected while trying to collect rows", err)
+	}
+
+	var id = rawMap[0].ID
+	var title = rawMap[0].Title
+
+	if err != nil {
+		t.Errorf("error '%s' was not expected while trying to scan row", err)
+	}
+
+	if id != 5 {
+		t.Errorf("expected mocked id to be 5, but got %d instead", id)
+	}
+
+	if title != "hello world" {
+		t.Errorf("expected mocked title to be 'hello world', but got '%s' instead", title)
+	}
+
+	if err := mock.ExpectationsWereMet(); err != nil {
+		t.Errorf("there were unfulfilled expectations: %s", err)
+	}
+}

